engine/src/main/java/org/destinationsol/ui/nui/NUIManager.java

```diff
@@ -0,0 +1,321 @@
+    }
+
+    /**
+     * Obtains the topmost screen (rendered on-top of all others)
+     * @return the topmost screen
+     */
+    public NUIScreenLayer getTopScreen() {
+        return uiScreens.peek();
+    }
+
+    /**
+     * Pushes a screen onto the UI stack.
+     * @param layer the screen to add
+     */
+    public void pushScreen(NUIScreenLayer layer) {
+        uiScreens.push(layer);
+
+        layer.setFocusManager(focusManager);
+        layer.setNuiManager(this);
+        layer.initialise();
+    }
+
+    /**
+     * Removes the topmost screen from the UI stack and returns it.
+     * @return the topmost screen
+     */
+    public NUIScreenLayer popScreen() {
+        if (!uiScreens.isEmpty()) {
+            uiScreens.peek().onRemoved();
+        }
+        return uiScreens.pop();
+    }
+
+    /**
+     * Removes a screen form the UI stack. It is no longer updated or drawn.
+     * @param screen the screen to remove
+     */
+    public void removeScreen(NUIScreenLayer screen) {
+        screen.onRemoved();
+        uiScreens.remove(screen);
+    }
+
+    /**
+     * States if a screen is currently present on the UI stack.
+     * @param screen the screen to search for
+     * @return true if the screen is currently on the UI stack, otherwise false
+     */
+    public boolean hasScreen(NUIScreenLayer screen) {
+        return uiScreens.contains(screen);
+    }
+
+    /**
+     * States if a screen of a specified type is present on the UI stack.
+     * @param type the type of screen to search for
+     * @return true if a screen of the specified type is currently on the UI stack, otherwise false
+     */
+    public boolean hasScreenOfType(Class<? extends NUIScreenLayer> type) {
+        for (NUIScreenLayer screenLayer : uiScreens) {
+            if (screenLayer.getClass().isAssignableFrom(type)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Returns all of the UI screens currently on the UI stack.
+     * @return the UI stack
+     */
+    public Deque<NUIScreenLayer> getScreens() {
+        return uiScreens;
+    }
+
+    /**
+     * Returns the default {@link UISkin} for widgets.
+     * @return the default {@link UISkin}
+     */
+    public UISkin getDefaultSkin() {
+        return skin;
+    }
+
+    /**
+     * Re-sizes the current canvas to a particular width and height. This is not always the same as the game's current
+     * rendering resolution.
+     * @param width the width to use
+     * @param height the height to use
+     */
+    public void resize(int width, int height) {
+        canvasRenderer.resize(width, height);
+    }
+
+    /**
+     * Sets the UI scale to a specified value. This will internally render the UI at specified scale and up-scale it
+     * to the target resolution.
+     * @param scale the new UI scale to use
+     */
+    public void setUiScale(float scale) {
+        canvas.setUiScale(1.0f / scale);
```

Comment:
To me it makes sense to have bigger elements when the scale is bigger. Is there any reason you choose to invert this?

Reply:
It was inverted mostly to reflect what was occurring internally (in [`LibGDXCanvasRenderer`](https://github.com/MovingBlocks/TeraNUI/blob/e1f7918018061864c99a72ef291f76299cb754d8/nui-libgdx/src/main/java/org/terasology/nui/backends/libgdx/LibGDXCanvasRenderer.java)), where the UI is being rendered at a resolution that is `scale*actual` and then scaled to match the actual resolution. From a visual perspective though, if the internal render resolution is less than the actual output resolution then the UI would appear larger. If the internal render resolution is greater then the UI would appear smaller.

I've changed it now though so that the values are more intuitive from a user-facing perspective (e.g. how it visibly affects the output).